index.js

```diff
@@ -64,7 +65,9 @@ function mergeReports(reports) {
 function runEslint(paths, opts) {
 	var config = optionsManager.buildConfig(opts);
 	var engine = new eslint.CLIEngine(config);
-	return engine.executeOnFiles(paths, config);
+	var report = engine.executeOnFiles(paths, config);
+	report.results = opts.quiet ? eslint.CLIEngine.getErrorResults(report.results) : report.results;
```

Comment:
Add a new `processReport(report, opts)` function that contains this common code:

```js
function processReport(report, opts) {
  report.results = ....
  return report;
}